### PR TITLE
fix: add `type` field in `boards` table

### LINKED ISSUE #2801: [Feature][Issue Tracking] Identify scrum and non-scrum board
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

1. I propose that we can add a field into "board" object to identify the type of different boards.
2. Also, when I pick up a data scope of Jira connection or some other tools within blueprint on Config UI, I want to know what kind of boards they are as well.

### Use case

In practice, a Scrum board and a Normal board represent two different agile patterns, and when analyzing delivery performance, we will use different data aggregate method.

E.g. 
- Since a scrum board normally have sprints to collect data, we will gather, aggregate, and compare issue data by each sprints. 
- When a team applies Kanban agile practice, they will pick up issue from backlog periodically, like every fortnight. Therefore, gathering, aggregating, and comparing issue date by fortnight is a natural choice.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thenicetgp**
# Summary
Add field type to domain layer boards table
Convert jira_board.type to board.type
Other plugins can't find the corresponding fields so  their board.type not be filled.

### Does this close any open issues?
closes #2801 

### Screenshots
_tool_jira_boards table
Screen Shot 2022-08-30 at 22 48 54

boards table
Screen Shot 2022-08-30 at 22 49 02

### CODE REVIEW

File: models/migrationscripts/20220830_add_type_field_in_board.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+package migrationscripts
+
+import (
+	"context"
+	"github.com/apache/incubator-devlake/models/migrationscripts/archived"
+	"gorm.io/gorm"
+	"time"
+)
+
+type AddFieldTask20220830 struct {
```
Comment (mindlesscloud) [inline models/migrationscripts/20220830_add_type_field_in_board.go:27]: an unexported name would be better

File: models/migrationscripts/20220830_add_type_field_in_board.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	archived.DomainEntity
+	Name        string `gorm:"type:varchar(255)"`
+	Description string
+	Url         string `gorm:"type:varchar(255)"`
+	CreatedDate *time.Time
+	Type string
```
Comment (mindlesscloud) [inline models/migrationscripts/20220830_add_type_field_in_board.go:33]: could we specify the data type as VARCHAR?

Comment (thenicetgp) [reply models/migrationscripts/20220830_add_type_field_in_board.go:33]: ok, changed it.

File: models/migrationscripts/20220830_add_type_field_in_board.go (hunk 1 above)
Comment (thenicetgp) [reply models/migrationscripts/20220830_add_type_field_in_board.go:27]: I have seen some exported name in some migration scripts. When do we need unexported name?

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**